blockprod/src/detail/tests.rs

```diff
@@ -1019,14 +1018,25 @@ mod produce_block {
                 vec![kernel_input_utxo.clone()],
             );
 
+            let easy_pos_config = PoSChainConfig::new(
```

Comment:
isn't it the same as `create_unittest_pos_config`? Difficulty is already at MAX by default. No sure that I understand why it's necessary

Reply:
The last value is different.

Reply:
oh right I confused it with another PerThousand value. But anyway does it make a difference? Could you give me a seed where you see it so I can check as well?

Reply:
I don't have any tests that I recorded, but I did have that test run over 60s while debugging

Reply:
still not sure that this change is necessary. I'd fix other problem and then look if retargeting code has an issues.

Reply:
Now that it's no longer failing with a Chainstate-less JobManager from `2f8f466`, I've fixed it to get the worst case:

```
modified   blockprod/src/detail/tests.rs                                                            
@@ -999,7 +999,7 @@ mod produce_block {                                                             
             )                                                                                      
         };                                                                                         
                                                                                                    
-        let blocks_to_generate = rng.gen_range(100..=1000);                                        
+        let blocks_to_generate = 1000;                                                             
                                                                                                    
         let override_chain_config = {                                                              
             let genesis_block = Genesis::new(                                                      
@@ -1048,14 +1048,14 @@ mod produce_block {                                                         
             let mut next_height_consensus_change = 1;                                              
                                                                                                    
             while next_height_consensus_change < blocks_to_generate {                              
-                let next_consensus_type = rng.gen_range(0..consensus_types.len());                 
+                let next_consensus_type = 2;                                                       
                                                                                                    
                 randomized_net_upgrades.push((                                                     
                     BlockHeight::new(next_height_consensus_change),                                
                     UpgradeVersion::ConsensusUpgrade(consensus_types[next_consensus_type].clone()),
                 ));                                                                                
                                                                                                    
-                next_height_consensus_change += rng.gen_range(1..50);                              
+                next_height_consensus_change += 1000;                                              
             }                                                                                      
``` 

- IgnoreConsensus: ~1s
- PoW: ~1s
- PoS: ~10s

However, if I change that last `PosChainConfig` value from `PerThousand::new(1)` back to `PerThousand::new(100)` like in `create_unittest_pos_config()`, I get the following:

- IgnoreConsensus: ~1s
- PoW: ~1s
- PoS: ~160s

We could revert it back to `PerThousand::new(100)`, but every now and again we might land on the worst case scenario and it will break CI.

Once last experiment - I just tried bumping it a bit higher to `PerThousand::new(10)`, and I'm getting:

- IgnoreConsensus: ~1s
- PoW: ~1s
- PoS: ~13s

Reply:
interesting, ok let's merge it and I'll look into it

Reply:
Yep cool. Just waiting on @TheQuantumPhysicist to discuss the Chainstate-less commit.